llvm-project/stage2.cmake

```diff
@@ -1,22 +1,74 @@
-# TODO: comment every one of these.
+# Please try to keep these cmake variables alphabetically sorted.
+
+# Enable optimizations, as opposed to a debug build.
 set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+# Use stage1's clang to bootstrap stage2, as opposed to Alpine's clang (which
+# shouldn't be installed on the system at this point, but would be installed to
+# /usr/bin/clang).
 set(CMAKE_CXX_COMPILER "/usr/local/bin/clang++" CACHE FILEPATH "")
+
+# Use libc++ from stage1.
+# We could not do this in stage1 because Alpine doesn't distribue libc++ ATM.
+# TODO: does clang cmake not have a CLANG_DEFAULT_* flag for setting this?
 set(CMAKE_CXX_FLAGS "--stdlib=libc++" CACHE STRING "")
+
+# See above comment for CMAKE_CXX_COMPILER.
 set(CMAKE_C_COMPILER "/usr/local/bin/clang" CACHE FILEPATH "")
+
+# Use libunwind from stage1.
+# Statically link resulting executable.
+# TODO: why is -lc++abi necessary for static link?
 set(CMAKE_EXE_LINKER_FLAGS "--unwindlib=libunwind -static -lc++abi" CACHE STRING "")
 set(CMAKE_SHARED_LINKER_FLAGS "--unwindlib=libunwind" CACHE STRING "")
+
+# Set the default target triple to match the host.
+# TODO: passing in the value of $(clang -print-multiarch) causes failures.
+# It seems that alpine clang's default target triple is x86_64-linux-gnu.
+# Perhaps missing alpine in the triple causes some incompatibility?
 set(LLVM_DEFAULT_TARGET_TRIPLE x86_64-alpine-linux-musl CACHE STRING "")
+
+# Use libc++ from stage1.
+# TODO: is CMAKE_CXX_FLAGS still necessary if this is set?
```

Comment:
I believe so, this doesn't impact the build but rather how the resulting binary is built IIRC

Reply:
```diff
-# We could not do this in stage1 because Alpine doesn't distribue libc++ ATM.
-# TODO: does clang cmake not have a CLANG_DEFAULT_* flag for setting this?
+# We could not set CLANG_DEFAULT_CXX_STDLIB in stage1 because Alpine doesn't
+# distribue libc++ ATM.
+# Even though we later will set LLVM_ENABLE_LIBCXX, cmake will fail thinking
+# that "The C++ compiler is not able to compile a simple test program." As with
+# stag1, this is an unfortunate effect of Alpine packaging the stdlibc++
+# headers in the g++ package, rather than the libstdc++ package. In stage1, we
+# used those headers; in stage2 we do not.
```
is what I'm updating this comment to.